library/src/lib/menu/menu.component.ts

```diff
@@ -1,16 +1,147 @@
-import { Component, HostBinding, ViewEncapsulation } from '@angular/core';
+import {
+    AfterContentInit,
+    Component,
+    ContentChildren, EventEmitter,
+    Input, OnDestroy, Output, QueryList,
+    ViewEncapsulation
+} from '@angular/core';
+import { MenuGroupComponent } from './menu-group.component';
+import { MenuListDirective } from './menu-list.directive';
+import { MenuItemDirective } from './menu-item.directive';
+import { Subscription } from 'rxjs';
 
 /**
- * The component that represents a menu.
+ * The component that represents a menu. Provides some keyboard event default handler.
  */
 @Component({
     selector: 'fd-menu',
     templateUrl: './menu.component.html',
-    encapsulation: ViewEncapsulation.None
+    encapsulation: ViewEncapsulation.None,
+    host: {
+        'class': 'fd-menu'
+    }
 })
-export class MenuComponent {
+export class MenuComponent implements AfterContentInit, OnDestroy {
 
     /** @hidden */
-    @HostBinding('class.fd-menu')
-    fdMenuClass: boolean = true;
+    @ContentChildren(MenuGroupComponent) menuGroup: QueryList<MenuGroupComponent>;
+
+    /** @hidden */
+    @ContentChildren(MenuListDirective) menuList: QueryList<MenuListDirective>;
+
+    /** Event thrown always, when item link is clicked */
+    @Output() itemClicked = new EventEmitter<number>();
+
+    /** Whether user wants to remove default keyboard handling */
+    @Input() disableKeydownHandling: boolean = false;
+
+    /** Whether user wants to define own function, that is supposed to be called, when focus escape before list*/
+    @Input() focusEscapeBeforeList: Function;
+
+    /** Whether user wants to define own function, that is supposed to be called, when focus escape after list*/
+    @Input() focusEscapeAfterList: Function;
+
+    private onItemKeyDownSubscription: Subscription[];
+    private onItemClickSubscription: Subscription[];
+
+    /** @hidden */
+    public ngAfterContentInit(): void {
+        this.refreshList();
+        this.menuList.forEach(list => list.listRefresh.subscribe(() => this.refreshList()));
+        this.menuGroup.forEach(list => list.menuList.listRefresh.subscribe(() => this.refreshList()));
+    }
+
+    /** @hidden */
+    public ngOnDestroy(): void {
+        this.onItemKeyDownSubscription = [];
```

Comment:
For this case, I think it's better to have a subject called when the component is destroyed, and have every dependent subscription takeUntil(destroy).

see https://github.com/SAP/fundamental-ngx/blob/feat/dropdown/library/src/lib/select/select.component.ts#L120

and https://github.com/SAP/fundamental-ngx/blob/feat/dropdown/library/src/lib/select/select.component.ts#L71

and https://github.com/SAP/fundamental-ngx/blob/feat/dropdown/library/src/lib/select/select.component.ts#L112